spec/consensus/light-client/detection.md

```diff
@@ -1,3 +1,459 @@
-# Detection
+***This is the beginning of an unfinished draft. Comments are welcome!***
+
+# Failure detector
+
+A failure detector is a process that gets as input a header with some height *h*, connects to different Tendermint full nodes, requests the header of height *h* from them, and then cross-checks the headers and the input header.
+
+There are two foreseeable use cases:
+
+1) strengthen the light client: If a light client accepts a header *hd* (after performing skipping or sequential verification), it can use the failure detector to probe the system for conflicting headers and increase the trust in *hd*. Instead of communicating with a single full node, communicating with several full nodes shall increase the likelihood to be aware of a fork in case there is one.
+
+2) to support fork accountability: In the case when more than 1/3 of the voting power is held by faulty validators, faulty nodes may generate two conflicting headers for the same height. The goal of the failure detector is to learn about the conflicting headers by probing different full nodes. Once a failure detector has two conflicting headers, these headers are evidence of misbehavior. A natural extension is to use the failure detector within a monitor process (on a full node) that calls the failure detector on a sample (or all) headers (in parallel). (If the sample is chosen at random, this adds a level of probabilistic reasoning.) If conflicting headers are found, they are evidence that can be used for punishing processes.
+
+In this document we will focus onn strengthening the light client, and leave other uses of the failure detection mechanism (e.g., when run on a full node) to the future.
+
+This document refers to the light client ADR  [[lightclient]]. If not
+familiar with the light client, the ADR gives a good overview.
+
+## Context of this document
+
+The light client verification specification [[verification]] is
+designed for the Tendermint failure model (1/3 assumption)
+[TMBC-FM-2THIRDS]. It is safe under this assumption, and live
+if it can reliably and timely communicate with a correct full node. If
```

Comment:
I agree that it is not precise here. We have to make sure it is precise in [[verification]], while here it is just an explanation and it might be precise enough for that prupose. What do you think?